integration/tests/cook/test_basic.py

```diff
@@ -211,11 +212,19 @@ def test_mea_culpa_retries(self):
 
     def test_compute_cluster(self):
         settings_dict = util.settings(self.cook_url)
-        # For now, we only have one compute cluster. This could be wrong with future
-        # refactors, but for now, this is the only compute cluster we have, so we should
-        # expect it.
         expected_compute_cluster = settings_dict['compute-clusters'][0]['config']['compute-cluster-name']
-        expected_mesos_framework = settings_dict['compute-clusters'][0]['config']['framework-id']
+        compute_cluster_factory_fn = settings_dict['compute-clusters'][0]['factory-fn']
```

Comment:
This smells like something we should refactor out. 
def cook_uses_kubernetes(....)
def cook_users_mesos(...)

Reply:
Done